### ISSUE TITLE
End of list (.) breaks indentation scope

### ISSUE THREAD

**Issue by theKBro** [doxygen/doxygen#8163]
**Describe the bug**
After ending a sub sub item list with a (.)
The indentation for the "More text for the first sub item" is broken!

**Expected behavior**
Keep indentation of sub item.

**To Reproduce**
Example from official "Lists" documentation:
```
/**
 * Text before the list
 * - list item 1
 *   - sub item 1
 *     - sub sub item 1
 *     - sub sub item 2
 *     . 
 *     The dot above ends the sub sub item list.
 *
 *     More text for the first sub item
 *   .
 *   The dot above ends the first sub item.
 *
 *   More text for the first list item
 *   - sub item 2
 *   - sub item 3
 * - list item 2
 * .
 * More text in the same paragraph.
 *
 * More text in a new paragraph.
 */
```

**Result Screenshot**
![image](https://user-images.githubusercontent.com/14205420/98354247-b6a15200-2020-11eb-9cd5-15d8e326e69b.png)

**Version**
doxygen 1.8.20
Windows 10, 64 bit

**Comment by theKBro** [comment]
additionally there are following warnings during generation:
```
foo.txt:12: warning: End of list marker found without any preceding list items
foo.txt:17: warning: Invalid list item found
foo.txt:19: warning: End of list marker found without any preceding list items
```

**Comment by albert-github** [comment]
As far as I could reconstruct , the problem surfaces the first time after the commit:
```
Commit: fdefd3091128296d8f572f7daba6d838de24bf4b [fdefd30]
Date: Monday, July 27, 2020 2:15:58 PM
Additional tweaks to get markdown tables inside ALIASES work

- Changed \_linebr to \ilinebr
- \ilinebr is now also passed to doctokenizer
- Also fixes issue #7493 regarding \snippet inside markdown tables and
  dealing with wrong line on issues detected by docparser after a
  markdown table.
- Added function tracing to markdown (enabled with -d markdown in a
  debug build)
```
the previous commit:
```
Commit: ab7dfc4d6de5a65946e5c0e54eab6fe21e7e44eb [ab7dfc4]
Date: Saturday, July 25, 2020 7:55:01 PM
```
Looks still OK.

It looks like the complete "stack" in respect to lists is terminated as soon as the first `.` is encountered, though I don't see direct evidence of this in the changed code (or another problem).

**Comment by theKBro** [comment]
seems to be fixed in master but is still an issue in 1.9.1

**Comment by albert-github** [comment]
Thanks for the update, problem is indeed still present in version 1.9.1 but is gone in the current master.

Problem is fixed but not yest released (please don't close the issue as this will be done at the moment of an official release)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
